controlpanel/api/aws.py

```diff
@@ -290,17 +290,19 @@ def remove_resource(self, arn, sid):
                 if resource.startswith(arn):
                     resources.remove(resource)
 
-    def grant_access(self, bucket_arn, access_level):
+    def grant_object_access(self, arn, access_level):
         other_level = 'readwrite' if access_level == 'readonly' else 'readonly'
 
-        self.add_resource(f'{bucket_arn}/*', access_level)
-        self.remove_resource(f'{bucket_arn}/*', other_level)
-        self.add_resource(bucket_arn, 'list')
+        self.add_resource(f'{arn}/*', access_level)
+        self.remove_resource(arn, other_level)
```

Comment:
Is it necessary to call `self.remove_resource` here? (genuine question) I see that the client code calls `policy.revoke_access(bucket_arn)` already before looping through the `path_arns` 🤷‍♂

Reply:
I was just thinking about this. I think it's superfluous for the reason you mentioned, so it can be removed.